### PR TITLE
Updated system info and extern packages

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

File: sunpy/util/sysinfo.py [hunk 1]
```diff
@@ -1,149 +1,106 @@
...
     """
-    sys_prop = get_sys_dict()
+    base_reqs = get_distribution("sunpy").requires()
+    base_reqs = {base_req.name.lower() for base_req in base_reqs}
+    extra_reqs = get_distribution("sunpy").requires(extras=["all"])
+    extra_reqs = sorted({extra_req.name.lower() for extra_req in extra_reqs}.difference(base_reqs))
+
+    missing_packages, installed_packages = find_dependencies(package="sunpy", extras=["all"])
+    extra_prop = {'System': platform.system(),
+                  'Arch': f"{platform.architecture()[0]}, ({platform.processor()})",
```
Comment (Cadair) [inline sunpy/util/sysinfo.py:68]:
```suggestion
                  "Arch": f"{platform.architecture()[0]}, ({platform.processor()})",
```

File: sunpy/util/sysinfo.py [hunk 2]
```diff
@@ -1,149 +1,106 @@
...
+    Prints ones' system info in an "attractive" fashion.
     """
-    sys_prop = get_sys_dict()
+    base_reqs = get_distribution("sunpy").requires()
+    base_reqs = {base_req.name.lower() for base_req in base_reqs}
+    extra_reqs = get_distribution("sunpy").requires(extras=["all"])
+    extra_reqs = sorted({extra_req.name.lower() for extra_req in extra_reqs}.difference(base_reqs))
+
+    missing_packages, installed_packages = find_dependencies(package="sunpy", extras=["all"])
+    extra_prop = {'System': platform.system(),
```
Comment (Cadair) [inline sunpy/util/sysinfo.py:67]:
```suggestion
    extra_prop = {"System": platform.system(),
```

**Review by Cadair — APPROVED** [review]

File: sunpy/util/sysinfo.py [hunk 3]
```diff
@@ -1,149 +1,106 @@
...
+from pkg_resources import get_distribution
 
 from sunpy.extern.distro import linux_distribution
 
-__all__ = ['get_sys_dict', 'system_info']
+__all__ = ['system_info', 'find_dependencies', 'missing_dependencies_by_extra']
 
 
-def get_sys_dict():
+def find_dependencies(package="sunpy", extras=None) -> Tuple[dict, dict]:
```
Comment (Cadair) [inline sunpy/util/sysinfo.py:17]:
```suggestion
def find_dependencies(package="sunpy", extras=None):
```

File: sunpy/util/sysinfo.py [hunk 4]
```diff
@@ -1,149 +1,106 @@
...
+    for requirement in requires:
+        try:
+            package = get_distribution(requirement)
+            installed_requirements[package.project_name.lower()] = package.version
+        except Exception:
+            missing_requirements[requirement.name.lower()] = f"Missing, need {requirement}"
+    return missing_requirements, installed_requirements
+
+
+def missing_dependencies_by_extra(package="sunpy", exclude_extras=None) -> dict:
```
Comment (Cadair) [inline sunpy/util/sysinfo.py:39]:
```suggestion
def missing_dependencies_by_extra(package="sunpy", exclude_extras=None):
```

**Review by Cadair — APPROVED** [review]

File: sunpy/util/sysinfo.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/util/sysinfo.py:17]: TOO LATE

File: sunpy/util/sysinfo.py (hunk 4 above)
Comment (nabobalis) [reply sunpy/util/sysinfo.py:39]: THE TYPING BEGINS

**Pull request merged**